publication-rest/src/main/java/no/unit/nva/publication/update/PublishingRequestResolver.java

```diff
@@ -162,27 +151,22 @@ private void updateFilesForApproval(
                         updatePublishingRequest(oldImage, newImage, publishingRequestCase));
     }
 
-    private Set<File> prepareFilesForApproval(Publication oldImage, Publication newImage) {
-        var filesForApproval = getFilesForApproval(oldImage, newImage);
-        removeFilesIfFileDoesNotExists(newImage, filesForApproval);
-        return filesForApproval;
-    }
-
-    private List<File> getNewPendingFiles(Publication oldImage, Publication newImage) {
-        var existingPendingFiles = getPendingFiles(oldImage);
-        var newPendingFiles = getPendingFiles(newImage);
-        newPendingFiles.removeIf(existingPendingFiles::contains);
-        return newPendingFiles;
+    private Stream<File> prepareFilesForApproval(Publication oldImage, Publication newImage) {
+        var filesForApproval = getNewPendingFiles(oldImage, newImage);
+        return removeFilesIfFileDoesNotExists(newImage, filesForApproval);
     }
 
-    private Set<File> getFilesForApproval(Publication oldImage, Publication newImage) {
-        return new HashSet<>(getNewPendingFiles(oldImage, newImage));
+    private Stream<File> getNewPendingFiles(Publication oldImage, Publication newImage) {
+        var existingPendingFiles = getPendingFiles(oldImage).toList();
+        return getPendingFiles(newImage)
+                   .filter(file -> !existingPendingFiles.contains(file))
+                   .distinct();
     }
 
     private void updatePublishingRequest(Publication oldImage, Publication newImage,
                                          PublishingRequestCase publishingRequest) {
-        var updatedFilesForApproval = prepareFilesForApproval(oldImage, newImage);
-        removeFilesIfFileDoesNotExists(newImage, updatedFilesForApproval);
+        var files = prepareFilesForApproval(oldImage, newImage);
+        var updatedFilesForApproval = removeFilesIfFileDoesNotExists(newImage, files).collect(Collectors.toSet());
```

Comment:
Nice, this line is not needed longer:
`        var updatedFilesForApproval = removeFilesIfFileDoesNotExists(newImage, files).collect(Collectors.toSet());`